lib/util/trip/trip_card_state.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/material.dart';
+import 'package:motis_mitfahr_app/util/trip/trip.dart';
+import 'package:motis_mitfahr_app/util/trip/trip_card.dart';
+import 'package:timelines/timelines.dart';
+
+import '../custom_timeline_theme.dart';
+import '../locale_manager.dart';
+
+abstract class TripCardState<T extends TripCard> extends State<T> {
+  Trip? trip;
+
+  BorderRadius cardBorder = const BorderRadius.only(
+    bottomRight: Radius.circular(10),
+    topRight: Radius.circular(10),
+  );
+
+  FixedTimeline buildRoute(context) {
+    return FixedTimeline(
+      theme: CustomTimelineTheme.of(context),
+      children: [
+        TimelineTile(
+          contents: Padding(
+            padding: const EdgeInsets.all(16.0),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(
+                  '${localeManager.formatTime(trip!.startTime)}  ${trip!.start}',
+                  style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 16),
+                ),
+              ],
+            ),
+          ),
+          node: const TimelineNode(
+            indicator: CustomOutlinedDotIndicator(),
+            endConnector: CustomSolidLineConnector(),
+          ),
+        ),
+        TimelineTile(
+          contents: Padding(
+            padding: const EdgeInsets.all(16),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                SizedBox(
+                  child: Row(
+                    children: [
+                      const Icon(Icons.access_time_outlined),
+                      const SizedBox(width: 4),
+                      Text(
+                          "${trip!.duration.inHours.toString().padLeft(2, "0")}:${(trip!.duration.inMinutes % 60).toString().padLeft(2, "0")}"),
```

Comment:
can you move this to locale manager? `formatDuration`
And it should probably have a parameter to control whether the hours are padded. Here for the duration it makes sense, but I don't think it makes sense for the added duration of a ride